### ISSUE TITLE
GIL deadlock in PyDaftFile methods when reading S3 files under concurrent scan tasks

### ISSUE THREAD

**Issue by zhouwenjiajia** [Eventual-Inc/Daft#7256]
### Describe the bug

In Ray distributed mode, `PyDaftFile`'s `#[pymethods]` (`_from_file_reference`, `read`, `seek`, `_supports_range_requests`, `__enter__`) hold the Python GIL while calling `block_within_async_context`, which blocks via `rx.recv()` waiting for async S3 I/O on DAFTIO tokio worker threads. When concurrent scan tasks need the GIL on the same DAFTIO threads (via `Python::attach`), all threads deadlock permanently — no error, no timeout, no recovery.

Confirmed with py-spy (BackgroundEventLoop thread `active+gil` stuck at `File.open()`), kernel thread stacks (all 8 DAFTIO threads on `futex_wait`), and `ss` (0 TCP connections to S3 — the request was never sent).

The `exists()` method in the same file (`python.rs:102`) correctly uses `py.detach()`. The 5 affected methods missed this pattern.

### To Reproduce

**Environment**: Ray distributed mode, `os.cpu_count()` >= 8, S3-compatible storage with latency >= 50ms.

```python
import daft

daft.set_runner_ray()
daft.set_planning_config(default_io_config=daft.io.IOConfig(
    s3=daft.io.S3Config(
        endpoint_url="https://your-s3-compatible-endpoint",
        region_name="your-region",
        key_id="...",
        access_key="...",
    )
))

df = daft.read_video_frames("s3://bucket/videos/", image_height=640, image_width=640)
df.count_rows()  # hangs indefinitely after processing ~19K frames
```

**All three conditions required to trigger:**
1. Multiple concurrent `run_plan` tasks per worker (Ray distributed mode)
2. S3 latency >= ~50ms (cross-region S3, MinIO, GCS, Azure Blob, Alibaba Cloud OSS, etc.)
3. `os.cpu_count()` >= 8 on the worker machine (DAFTIO threads = `min(8, cpu_count)`)

AWS g6.xlarge (4 vCPU, same-AZ S3 < 1ms) does not trigger due to low thread count and latency.

### Expected behavior

The pipeline should complete without deadlocking, regardless of S3 latency or DAFTIO thread count.

### Component(s)

Cloud Storage (S3/GCS/Azure), Data Sources & Sinks, Distributed Runner (flotilla)

### Additional context

**Deadlock cycle:**
```
① BackgroundEventLoop: holds GIL → File.open() → rx.recv() blocks (no py.detach())
② 5 DAFTIO threads: queue.get() waiting for data from ①
③ 3 DAFTIO threads: Python::attach waiting for GIL held by ①
④ S3 task: needs a DAFTIO thread → all 8 blocked (②+③) → never executes
→ rx.recv() never returns → GIL never released → permanent deadlock
```

**Diagnostic evidence:**
- `py-spy dump`: BackgroundEventLoop `(active+gil)` at `File.open()`, 5 threads at `queue.get()`, 3 threads `(active)` no Python stack (waiting for GIL in Rust `Python::attach`)
- Kernel stacks: `DAFTIO-0` through `DAFTIO-7` all on `futex_wait` for 24+ hours
- `ss -tnopi | grep 443`: 0 connections to S3

**Affected code** (`src/daft-file/src/python.rs`):
- `PyFileReference::__enter__` (line 33)
- `PyDaftFile::_from_file_reference` (line 150)
- `PyDaftFile::read` (line 155)
- `PyDaftFile::seek` (line 193)
- `PyDaftFile::_supports_range_requests` (line 260)

**Proposed fix**: Wrap blocking calls in `py.detach()`, same pattern as `exists()` at line 102. For `read()`/`seek()`, use `cursor.take()` + restore to satisfy Rust ownership.

**Verified**: 24+ hour deadlock → 9.6 minutes after fix.
